EXPLAINER.md

```diff
@@ -50,24 +50,94 @@ await port.open();
 
 At this point the `readable` and `writable` attributes are populated with a [`ReadableStream`](https://streams.spec.whatwg.org/#rs-class) and [`WritableStream`](https://streams.spec.whatwg.org/#ws-class) that can be used to receive data from and send data to the connected device.
 
-In this example we assume a device implementing a protocol inspired by the [Hayes command set](https://en.wikipedia.org/wiki/Hayes_command_set). Since commands are encoded in ASCII a [`TextEncoderStream`](https://encoding.spec.whatwg.org/#interface-textencoderstream) and [`TextDecoderStream`](https://encoding.spec.whatwg.org/#interface-textdecoderstream) are attached to the port's streams in order to translate the `Uint8Array`s used by the `SerialPort`'s streams into strings.
+In this example we assume a device implementing a protocol inspired by the [Hayes command set](https://en.wikipedia.org/wiki/Hayes_command_set). Since commands are encoded in ASCII a [`TextEncoder`](https://encoding.spec.whatwg.org/#interface-textencoder) and [`TextDecoder`](https://encoding.spec.whatwg.org/#interface-textdecoder) are used to translate the `Uint8Array`s used by the `SerialPort`'s streams to and from strings.
 
 ```javascript
-let encoder = new TextEncoderStream();
-encoder.readable.pipeTo(port.writable);
-let writer = encoder.writable.getWriter();
+const encoder = new TextEncoder();
+const writer = encoder.writable.getWriter();
+writer.write(encoder.encode("AT"));
+
+const decoder = new TextDecoder();
+const reader = port.readable.getReader();
+const { value, done } = await reader.read();
+console.log(decoder.decode(value));
+// Expected output: OK
+```
+
+The readable and writable streams must be unlocked before the port can be closed.
+
+```javascript
+writer.releaseLock();
+reader.releaseLock();
+await port.close();
+```
+
+Rather than reading a single chunk from the stream code will often read continuously using a loop like this,
+
+```javascript
+const reader = port.readable.getReader();
+while (true) {
```

Comment:
See my suggestion of using `async for` instead in the other PR.

Reply:
I've never tried it before. I'd like to stick to simpler examples for the moment.